src/main/java/bio/terra/cli/service/WorkspaceManagerServiceGcp.java

```diff
@@ -692,6 +745,46 @@ public void deleteControlledGcpNotebookInstance(UUID workspaceId, UUID resourceI
         "Error deleting controlled GCP Notebook instance in the workspace.");
   }
 
+  /**
+   * Call the Workspace Manager POST
+   * "/api/workspaces/v1/{workspaceId}/resources/controlled/gcp/dataproc-clusters/{resourceId}"
+   * endpoint to delete a GCP Dataproc cluster as a controlled resource in the workspace.
+   *
+   * @param workspaceId the workspace to remove the resource from
+   * @param resourceId the resource id
+   * @throws SystemException if the job to delete the GCP Dataproc cluster fails
+   * @throws UserActionableException if the CLI times out waiting for the job to complete
+   */
+  public void deleteControlledGcpDataprocCluster(UUID workspaceId, UUID resourceId) {
+    ControlledGcpResourceApi controlledGcpResourceApi = new ControlledGcpResourceApi(apiClient);
+    String asyncJobId = UUID.randomUUID().toString();
+    DeleteControlledGcpDataprocClusterRequest deleteRequest =
+        new DeleteControlledGcpDataprocClusterRequest().jobControl(new JobControl().id(asyncJobId));
+    handleClientExceptions(
+        () -> {
+          // make the initial delete request
+          HttpUtils.callWithRetries(
+              () ->
+                  controlledGcpResourceApi.deleteDataprocCluster(
+                      deleteRequest, workspaceId, resourceId),
+              WorkspaceManagerService::isRetryable);
+
+          // poll the result endpoint until the job is no longer RUNNING
+          DeleteControlledGcpDataprocClusterResult deleteResult =
```

Comment:
How long does a delete take? is it a candidate for async op as well?

Reply:
delete is around the same as vertex notebook delete, I leave this one as sync